__tests__/auth.test.ts

```diff
@@ -128,17 +147,55 @@ describe('auth tests', () => {
     const id = 'packages';
     const username = 'USER';
     const password = '&<>"\'\'"><&';
+    const gpgPassphrase = 'PASSPHRASE';
 
-    expect(auth.generate(id, username, password)).toEqual(`
+    expect(auth.generate(id, username, password, gpgPassphrase)).toEqual(`
   <settings>
       <servers>
         <server>
           <id>${id}</id>
           <username>\${env.${username}}</username>
           <password>\${env.&amp;&lt;&gt;&quot;&apos;&apos;&quot;&gt;&lt;&amp;}</password>
         </server>
+        <server>
+          <id>gpg.passphrase</id>
+          <passphrase>\${env.${gpgPassphrase}}</passphrase>
+        </server>
       </servers>
   </settings>
   `);
   });
+
+  it('imports gpg private key', async () => {
+    const id = 'packages';
+    const username = 'USERNAME';
+    const password = 'PASSWORD';
+    const gpgPrivateKey = 'KEY CONTENTS';
+
+    await auth.configAuthentication(id, username, password, gpgPrivateKey);
+
+    expect(exec.exec).toHaveBeenCalledWith(
+      'gpg',
+      ['--import', '--batch', privateKeyFile],
+      {cwd: privateKeyDir}
+    );
+
+    expect(fs.existsSync(privateKeyDir)).toBe(false);
+  }, 100000);
+
+  it('does not import gpg private key when private key is not set', async () => {
+    const id = 'packages';
+    const username = 'USERNAME';
+    const password = 'PASSWORD';
+
+    await auth.configAuthentication(id, username, password);
+
+    expect(exec.exec).not.toHaveBeenCalledWith(
```

Comment:
I’d make this less precise and go with toHaveBeenCalled instead. If your exec call drifts and for example the cwd is passed incorrectly this would still pass.

Reply:
I opted for preciseness in the potential future case that someone else decides to call exec. But you're right, there are definitely drawbacks to this approach. I'll change it 👍

Reply:
Found another approach that allows a future case where exec is called for something else but also deals with argument drift.